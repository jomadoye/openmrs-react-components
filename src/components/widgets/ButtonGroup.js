import React from 'react';
import { ToggleButtonGroup, ToggleButton } from 'react-bootstrap';

const buttonStyle = {
  whiteSpace: "normal"
};

const ButtonGroup = ({ input, options }) => (

  <ToggleButtonGroup type="radio" justified={true} {...input}>
    { options.map( option =>
      <ToggleButton key={ option.uuid } value={option.uuid} style={ buttonStyle }>{ option.name }</ToggleButton>
    )}
  </ToggleButtonGroup>

);

export default ButtonGroup;
