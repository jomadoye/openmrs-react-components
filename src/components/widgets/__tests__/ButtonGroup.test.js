import React from 'react';
import renderer from 'react-test-renderer';
import ButtonGroup from '../ButtonGroup';

let props;
let mountedComponent;

const getComponent = () => {
  if (!mountedComponent) {
    mountedComponent = renderer.create(<ButtonGroup {...props} />);
  }
  return mountedComponent;
};

describe('Component: ButtonGroup', () => {
  beforeEach(() => {
    props = {
      input: {
        name: 'mockname'
      },
      options: [{
        name: 'mockName',
        uuid: '1234'
      }]
    };
    mountedComponent = undefined;
  });

  it('renders properly', () => {
    const component = getComponent();
    expect(component.toJSON()).toMatchSnapshot();
  });
});
