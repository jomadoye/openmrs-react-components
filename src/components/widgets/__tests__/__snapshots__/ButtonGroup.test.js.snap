// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ButtonGroup renders properly 1`] = `
<div
  className="btn-group btn-group-justified"
  data-toggle="buttons"
>
  <label
    className="btn btn-default"
    disabled={false}
    style={
      Object {
        "whiteSpace": "normal",
      }
    }
    type="button"
  >
    <input
      autoComplete="off"
      checked={false}
      disabled={false}
      name="mockname"
      onChange={[Function]}
      type="radio"
      value="1234"
    />
    mockName
  </label>
</div>
`;
